app/code/Magento/Quote/Model/QuoteRepository.php

```diff
@@ -203,13 +214,13 @@ public function delete(\Magento\Quote\Api\Data\CartInterface $quote)
      * @param int $identifier
      * @param int[] $sharedStoreIds
      * @throws NoSuchEntityException
-     * @return Quote
+     * @return CartInterface
      */
     protected function loadQuote($loadMethod, $loadField, $identifier, array $sharedStoreIds = [])
     {
-        /** @var Quote $quote */
-        $quote = $this->quoteFactory->create();
-        if ($sharedStoreIds) {
+        /** @var CartInterface $quote */
+        $quote = $this->cartFactory->create();
+        if ($sharedStoreIds && $quote instanceof Quote) {
```

Comment:
Maybe just check `method_exists('setSharedStoreIds')` ?

Reply:
@orlangur actually this method does not exist I believe, it is declared only in `Quote` class docblock. Maybe we should change this to `$quote instanceof \Magento\Framework\DataObject || method_exists($quote, 'setSharedStoreIds')`? Because in both situations method should probably be called.

Reply:
@orlangur please check current solution 😉

Reply:
Please grep for `share.*store` to find whether we get it somewhere.

It works and is placed in DocBlock due to `DataObject` providing magic method `__call`. I would use it for a legacy `Quote` only or would check with `method_exists` so it won't break any implementation just covering `CartInterface` but not extending `AbstractModel`.

Reply:
I haven't found any usage that could be broken using grep. Please check now, I changed DataModel to Quote @orlangur

Reply:
I meant remove this condition if no getter called but this is not true: https://github.com/magento/magento2/blob/2.3-develop/app/code/Magento/Quote/Model/ResourceModel/Quote.php#L63

Reply:
@orlangur what should I do with this condition then?